Fix `float_t` incompatibility with RzFloat merged in RzUtil in Rizin

```
 > [linux/amd64 stage-0 15/15] RUN cmake -DCMAKE_PREFIX_PATH=/tmp/***-install/usr -DCMAKE_INSTALL_PREFIX=/usr -B build && cmake --build build && DESTDIR=/tmp/***-install cmake --build build --target install:
rizinorg/rz-ghidra#44 194.1 In file included from /tmp/***-install/usr/include/librz/rz_util.h:41,
rizinorg/rz-ghidra#44 194.1                  from /tmp/***-install/usr/include/librz/rz_io.h:10,
rizinorg/rz-ghidra#44 194.1                  from /tmp/***-install/usr/include/librz/rz_analysis.h:14,
rizinorg/rz-ghidra#44 194.1                  from /tmp/rz-ghidra/src/analysis_ghidra.cpp:7:
rizinorg/rz-ghidra#44 194.1 /tmp/***-install/usr/include/librz/rz_util/rz_float.h:70:16: note: previous declaration as 'struct float_t'
rizinorg/rz-ghidra#44 194.1  typedef struct float_t {
rizinorg/rz-ghidra#44 194.1                 ^~~~~~~
rizinorg/rz-ghidra#44 196.5 make[2]: *** [CMakeFiles/analysis_ghidra.dir/build.make:102: CMakeFiles/analysis_ghidra.dir/src/analysis_ghidra.cpp.o] Error 1
rizinorg/rz-ghidra#44 196.5 make[1]: *** [CMakeFiles/Makefile2:161: CMakeFiles/analysis_ghidra.dir/all] Error 2
rizinorg/rz-ghidra#44 196.5 make: *** [Makefile:130: all] Error 2
```

This is an issue in rizin, we should not define `float_t` since it is part of c++:
https://cplusplus.com/reference/cmath/float_t/
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DRZ_GHIDRA_SLEIGHHOME_DEFAULT=\"/Users/florian/.local/lib/rizin/plugins/rz_ghidra_sleigh\" -Dasm_ghidra_EXPORTS -I/Users/florian/dev/rz-ghidra/ghidra/ghidra/Ghidra/Features/Decompiler/src/decompile/cpp -I/Users/florian/dev/rz-ghidra/third-party/pugixml/src -isystem /Users/florian/bin/prefix/rizin/include/librz -isystem /Users/florian/bin/prefix/rizin/include/librz/sdb -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -fPIC -std=gnu++11 -MD -MT CMakeFiles/asm_ghidra.dir/src/asm_ghidra.cpp.o -MF CMakeFiles/asm_ghidra.dir/src/asm_ghidra.cpp.o.d -o CMakeFiles/asm_ghidra.dir/src/asm_ghidra.cpp.o -c /Users/florian/dev/rz-ghidra/src/asm_ghidra.cpp
In file included from /Users/florian/dev/rz-ghidra/src/asm_ghidra.cpp:6:
In file included from /Users/florian/dev/rz-ghidra/src/SleighAsm.h:11:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/unordered_map:437:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__hash_table:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/cmath:308:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/math.h:300:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/math.h:44:19: error: typedef redefinition with different types ('float' vs 'float_t')
    typedef float float_t;
                  ^
/Users/florian/bin/prefix/rizin/include/librz/rz_util/rz_float.h:70:16: note: previous definition is here
typedef struct float_t {
               ^
1 error generated.
ninja: build stopped: subcommand failed.
```

--- a/librz/include/rz_util/rz_float.h
+++ b/librz/include/rz_util/rz_float.h
@@ -112,8 +112,8 @@ RZ_API RZ_OWN RzFloat *rz_float_sqrt(RZ_NONNULL RzFloat *n, RzFloatRMode mode);
 
 RZ_API RZ_OWN RzFloat *rz_float_trunc(RZ_NONNULL RzFloat *f);
 RZ_API RZ_OWN RzFloat *rz_float_abs(RZ_NONNULL RzFloat *f);
-RZ_API RZ_OWN RzFloat *rz_float_new_from_hex_as_f64(ut64 hex_value);
-RZ_API RZ_OWN RzFloat *rz_float_new_from_hex_as_f32(ut32 hex_value);
+RZ_API RZ_OWN RzFloat *rz_float_new_from_ut64_as_f64(ut64 value);
+RZ_API RZ_OWN RzFloat *rz_float_new_from_ut32_as_f32(ut32 value);
 RZ_API RZ_OWN char *rz_float_as_string(RZ_NULLABLE RzFloat *f);
 RZ_API RZ_OWN char *rz_float_as_bit_string(RZ_NULLABLE RzFloat *f);
 RZ_API RZ_OWN char *rz_float_as_hex_string(RZ_NULLABLE RzFloat *f, bool use_pad);

--- a/librz/include/rz_util/rz_float.h
+++ b/librz/include/rz_util/rz_float.h
@@ -10,7 +10,7 @@
 #define RZ_FLOAT_H
 #include <rz_types.h>
 
-typedef enum float_format_enum {
+typedef enum rz_float_format_enum {
 	/// basic IEEE 754 float format enums
 	/// ref : https://en.wikipedia.org/wiki/IEEE_754#Basic_and_interchange_formats
 	/// 1. IEEE binary representations, use binary digits to represent float. machine-friendly
@@ -28,7 +28,7 @@ typedef enum float_format_enum {
 	RZ_FLOAT_UNK ///< End of enums
 } RzFloatFormat;
 
-typedef enum float_format_info {
+typedef enum rz_float_format_info {
 	RZ_FLOAT_INFO_BASE, ///< base of float representation, 2 for binary, 10 for decimal representation
 	RZ_FLOAT_INFO_EXP_LEN, ///< info about width of exponent field, in bits
 	RZ_FLOAT_INFO_MAN_LEN, ///< info about width of mantissa field, in bits
@@ -36,7 +36,7 @@ typedef enum float_format_info {
 	RZ_FLOAT_INFO_BIAS ///< exponent bias
 } RzFloatInfo;
 
-typedef enum float_round_enum {
+typedef enum rz_float_round_enum {
 	RZ_FLOAT_RMODE_RNE, ///< rounding to nearest, ties to even
 	RZ_FLOAT_RMODE_RNA, ///< rounding to nearest, ties away
 	RZ_FLOAT_RMODE_RTP, ///< rounding towards positive
@@ -45,7 +45,7 @@ typedef enum float_round_enum {
 	RZ_FLOAT_RMODE_UNK ///< end
 } RzFloatRMode; ///< Rounding Mode
 
-typedef enum float_exception_enum {
+typedef enum rz_float_exception_enum {
 	RZ_FLOAT_E_INVALID_OP = 1, ///< Invalid operation
 	RZ_FLOAT_E_DIV_ZERO = 2, ///< Divide zero
 	RZ_FLOAT_E_OVERFLOW = 4, ///< overflow exception
@@ -58,7 +58,7 @@ typedef enum float_exception_enum {
  * quiet_NaN : A == 1, signaling_NaN : A == 0
  * PA-RISC and MIPS, use A as is_signal flag. Should reverse the case
  */
-typedef enum float_speciality_enum {
+typedef enum rz_float_speciality_enum {
 	RZ_FLOAT_SPEC_NOT = 0, ///< not a special num
 	RZ_FLOAT_SPEC_ZERO = 1, ///< zero float
 	RZ_FLOAT_SPEC_PINF = 2, ///< positive infinity
@@ -67,7 +67,7 @@ typedef enum float_speciality_enum {
 	RZ_FLOAT_SPEC_SNAN = 5, ///< Signaling NaN
 } RzFloatSpec;
 
-typedef struct float_t {
+typedef struct rz_float_t {
 	RzFloatFormat r; ///< An interpretation of bitvector
 	RzBitVector *s; ///< The bitvector of float
 	RzFloatException exception; ///< exception of float operations

--- a/librz/util/float/float.c
+++ b/librz/util/float/float.c
@@ -297,39 +297,48 @@ RZ_API RZ_OWN RzFloat *rz_float_new_from_double(double value) {
 /**
  * \brief create RzFloat by giving hex value, most used in writing testcases
  * ref : http://www.jhauser.us/arithmetic/TestFloat-3/doc/TestFloat-general.html
- * \param hex_value 32-bit/64-bit hex value to represent 32-bit/64-bit bitvector
+ * \param value 32-bit/64-bit value to represent 32-bit/64-bit bitvector (big endian)
  * \param format float format
  * \return new RzFloat
  */
-static RZ_OWN RzFloat *rz_float_new_from_hex(ut64 hex_value, RzFloatFormat format) {
-	if ((format == RZ_FLOAT_IEEE754_BIN_32) || (format == RZ_FLOAT_IEEE754_BIN_64)) {
-		RzFloat *ret = RZ_NEW(RzFloat);
+static RZ_OWN RzFloat *float_new_from_ut64(ut64 value, RzFloatFormat format) {
+	RzFloat *ret = NULL;
+	switch (format) {
+	case RZ_FLOAT_IEEE754_BIN_32:
+		/* fall-thru */
+	case RZ_FLOAT_IEEE754_BIN_64:
+		ret = RZ_NEW(RzFloat);
+		if (!ret) {
+			RZ_LOG_ERROR("float: Cannot allocate RzFloat\n");
+			break;
+		}
 		ret->r = format;
-		ret->s = rz_bv_new_from_ut64(rz_float_get_format_info(format, RZ_FLOAT_INFO_TOTAL_LEN), hex_value);
-		return ret;
-	} else {
+		ret->s = rz_bv_new_from_ut64(rz_float_get_format_info(format, RZ_FLOAT_INFO_TOTAL_LEN), value);
+		break;
+	default:
 		// could not carry hex value larger than ut64 max
-		rz_warn_if_reached();
-		return NULL;
+		RZ_LOG_ERROR("float: unsupported float format type %u\n", format);
+		break;
 	}
+	return ret;
 }
 
 /**
  * \brief create RzFloat by giving 64-bit hex value, most used in writing testcases
- * \param hex_value 64-bit hex_value
+ * \param value 64-bit value to represent 64-bit bitvector (big endian)
  * \return RzFloat-binary64
  */
-RZ_API RZ_OWN RzFloat *rz_float_new_from_hex_as_f64(ut64 hex_value) {
-	return rz_float_new_from_hex(hex_value, RZ_FLOAT_IEEE754_BIN_64);
+RZ_API RZ_OWN RzFloat *rz_float_new_from_ut64_as_f64(ut64 value) {
+	return float_new_from_ut64(value, RZ_FLOAT_IEEE754_BIN_64);
 }
 
 /**
  * \brief create RzFloat by giving 32-bit hex value, most used in writing testcases
- * \param hex_value 32-bit hex_value
+ * \param value 32-bit value to represent 32-bit bitvector (big endian)
  * \return RzFloat-binary32
  */
-RZ_API RZ_OWN RzFloat *rz_float_new_from_hex_as_f32(ut32 hex_value) {
-	return rz_float_new_from_hex(hex_value, RZ_FLOAT_IEEE754_BIN_32);
+RZ_API RZ_OWN RzFloat *rz_float_new_from_ut32_as_f32(ut32 value) {
+	return float_new_from_ut64(value, RZ_FLOAT_IEEE754_BIN_32);
 }
 
 /**

--- a/librz/util/float/float.c
+++ b/librz/util/float/float.c
@@ -118,8 +118,7 @@ RZ_IPI ut32 rz_float_get_format_info(RzFloatFormat format, RzFloatInfo which_inf
 	case RZ_FLOAT_IEEE754_DEC_64:
 	case RZ_FLOAT_IEEE754_DEC_128:
 	default:
-		RZ_LOG_ERROR("FORMAT NOT IMPLEMENTED YET");
-		rz_warn_if_reached();
+		RZ_LOG_ERROR("float: info: Unsupported format %u\n", format);
 		return 0;
 	}
 }
@@ -195,7 +194,7 @@ RZ_API RZ_OWN RzFloat *rz_float_dup(RZ_NONNULL RzFloat *f) {
 	rz_return_val_if_fail(f, NULL);
 	RzFloat *cp = RZ_NEW(RzFloat);
 	if (!cp) {
-		RZ_LOG_ERROR("Dup float failed")
+		RZ_LOG_ERROR("float: dup: Cannot allocate RzFloat\n");
 		return NULL;
 	}
 
@@ -260,12 +259,12 @@ RZ_API bool rz_float_set_from_double(RZ_NONNULL RzFloat *f, double value) {
 RZ_API RZ_OWN RzFloat *rz_float_new_from_single(float value) {
 	RzFloat *f = rz_float_new(RZ_FLOAT_IEEE754_BIN_32);
 	if (!f) {
-		RZ_LOG_ERROR("Failed to new a single float")
+		RZ_LOG_ERROR("float: Failed to allocate single-precision RzFloat\n");
 		return NULL;
 	}
 
 	if (!rz_float_set_from_single(f, value)) {
-		RZ_LOG_ERROR("Error in set float from single")
+		RZ_LOG_ERROR("float: Error in setting RzFloat from single-precision float\n");
 		rz_float_free(f);
 		return NULL;
 	}
@@ -281,12 +280,12 @@ RZ_API RZ_OWN RzFloat *rz_float_new_from_single(float value) {
 RZ_API RZ_OWN RzFloat *rz_float_new_from_double(double value) {
 	RzFloat *f = rz_float_new(RZ_FLOAT_IEEE754_BIN_64);
 	if (!f) {
-		RZ_LOG_ERROR("Failed to new a double float")
+		RZ_LOG_ERROR("float: Failed to allocate double-precision RzFloat\n");
 		return NULL;
 	}
 
 	if (!rz_float_set_from_double(f, value)) {
-		RZ_LOG_ERROR("Error in set float from double")
+		RZ_LOG_ERROR("float: Error in setting RzFloat from double-precision float\n");
 		rz_float_free(f);
 		return NULL;
 	}
@@ -620,7 +619,7 @@ static RZ_OWN RzFloat *fadd_mag(RZ_NONNULL RzFloat *left, RZ_NONNULL RzFloat *ri
 	RzBitVector *r_mantissa = get_man(right->s, right->r);
 
 	if (!l_exp_squashed || !r_exp_squashed || !l_mantissa || !r_mantissa) {
-		RZ_LOG_ERROR("Error when parsing rz-float")
+		RZ_LOG_ERROR("float: fadd: Error when parsing RzFloat\n");
 		return NULL;
 	}
 
@@ -778,7 +777,7 @@ static RZ_OWN RzFloat *fsub_mag(RZ_NONNULL RzFloat *left, RZ_NONNULL RzFloat *ri
 	RzBitVector *r_mantissa = get_man(right->s, right->r);
 
 	if (!l_exp_squashed || !r_exp_squashed || !l_mantissa || !r_mantissa) {
-		RZ_LOG_ERROR("Error when parsing rz-float")
+		RZ_LOG_ERROR("float: fsub: Error when parsing RzFloat\n");
 		rz_bv_free(l_exp_squashed);
 		rz_bv_free(r_exp_squashed);
 		rz_bv_free(l_mantissa);

--- a/test/unit/test_float.c
+++ b/test/unit/test_float.c
@@ -329,9 +329,9 @@ bool rz_float_abs_test(void) {
 }
 
 bool rz_float_new_from_hex_test() {
-	RzFloat *hex1 = rz_float_new_from_hex_as_f32(0xC00007EF);
+	RzFloat *hex1 = rz_float_new_from_ut32_as_f32(0xC00007EF);
 	RzFloat *expect1 = rz_float_new_from_single(-2.0004842f);
-	RzFloat *hex2 = rz_float_new_from_hex_as_f32(0x3DFFF7BF);
+	RzFloat *hex2 = rz_float_new_from_ut32_as_f32(0x3DFFF7BF);
 	RzFloat *expect2 = rz_float_new_from_single(0.12498426f);
 
 	mu_assert_true(is_equal_float(hex1, expect1), "new from hex 1");
@@ -347,26 +347,26 @@ bool rz_float_new_from_hex_test() {
 
 bool f32_ieee_fma_test(void) {
 	RzFloat *a1, *b1, *c1, *expect1, *z1;
-	a1 = rz_float_new_from_hex_as_f32(0x2B6C2D9D);
-	b1 = rz_float_new_from_hex_as_f32(0xCB800000);
-	c1 = rz_float_new_from_hex_as_f32(0x4C440D9E);
-	expect1 = rz_float_new_from_hex_as_f32(0x4C440D9E);
+	a1 = rz_float_new_from_ut32_as_f32(0x2B6C2D9D);
+	b1 = rz_float_new_from_ut32_as_f32(0xCB800000);
+	c1 = rz_float_new_from_ut32_as_f32(0x4C440D9E);
+	expect1 = rz_float_new_from_ut32_as_f32(0x4C440D9E);
 	z1 = rz_float_fma(a1, b1, c1, RZ_FLOAT_RMODE_RNE);
 	mu_assert_true(is_equal_float(expect1, z1), "Fused Mul Add test 1");
 
 	RzFloat *a2, *b2, *c2, *expect2, *z2;
-	a2 = rz_float_new_from_hex_as_f32(0xBD0134F8);
-	b2 = rz_float_new_from_hex_as_f32(0x3F7FFFFE);
-	c2 = rz_float_new_from_hex_as_f32(0xC1C800D3);
-	expect2 = rz_float_new_from_hex_as_f32(0xC1C8416D);
+	a2 = rz_float_new_from_ut32_as_f32(0xBD0134F8);
+	b2 = rz_float_new_from_ut32_as_f32(0x3F7FFFFE);
+	c2 = rz_float_new_from_ut32_as_f32(0xC1C800D3);
+	expect2 = rz_float_new_from_ut32_as_f32(0xC1C8416D);
 	z2 = rz_float_fma(a2, b2, c2, RZ_FLOAT_RMODE_RNE);
 	mu_assert_true(is_equal_float(expect2, z2), "Fused Mul Add test 2");
 
 	RzFloat *a3, *b3, *c3, *expect3, *z3;
-	a3 = rz_float_new_from_hex_as_f32(0x6F7FFF7C);
-	b3 = rz_float_new_from_hex_as_f32(0x3F1DD0B8);
-	c3 = rz_float_new_from_hex_as_f32(0x81000000);
-	expect3 = rz_float_new_from_hex_as_f32(0x6F1DD067);
+	a3 = rz_float_new_from_ut32_as_f32(0x6F7FFF7C);
+	b3 = rz_float_new_from_ut32_as_f32(0x3F1DD0B8);
+	c3 = rz_float_new_from_ut32_as_f32(0x81000000);
+	expect3 = rz_float_new_from_ut32_as_f32(0x6F1DD067);
 	z3 = rz_float_fma(a3, b3, c3, RZ_FLOAT_RMODE_RNE);
 	mu_assert_true(is_equal_float(expect3, z3), "Fused Mul Add test 3");
 
@@ -405,11 +405,11 @@ bool f32_ieee_fma_test(void) {
 }
 
 bool f32_ieee_round_test(void) {
-	RzFloat *a = rz_float_new_from_hex_as_f32(0xC00007EF);
-	RzFloat *b = rz_float_new_from_hex_as_f32(0x3DFFF7BF);
+	RzFloat *a = rz_float_new_from_ut32_as_f32(0xC00007EF);
+	RzFloat *b = rz_float_new_from_ut32_as_f32(0x3DFFF7BF);
 
-	RzFloat *expect_rne_rna_rtp_rtz = rz_float_new_from_hex_as_f32(0xBFF01062);
-	RzFloat *expect_rtn = rz_float_new_from_hex_as_f32(0xBFF01063);
+	RzFloat *expect_rne_rna_rtp_rtz = rz_float_new_from_ut32_as_f32(0xBFF01062);
+	RzFloat *expect_rtn = rz_float_new_from_ut32_as_f32(0xBFF01063);
 
 	RzFloat *rne = rz_float_add(a, b, RZ_FLOAT_RMODE_RNE);
 	RzFloat *rna = rz_float_add(a, b, RZ_FLOAT_RMODE_RNA);
@@ -563,9 +563,9 @@ bool f32_ieee_rem_test(void) {
 	rz_float_free(expect1);
 	rz_float_free(rem1);
 
-	RzFloat *a2 = rz_float_new_from_hex_as_f32(0xCBF83FFF);
-	RzFloat *b2 = rz_float_new_from_hex_as_f32(0x44801003);
-	RzFloat *expect2 = rz_float_new_from_hex_as_f32(0xC3F52F40);
+	RzFloat *a2 = rz_float_new_from_ut32_as_f32(0xCBF83FFF);
+	RzFloat *b2 = rz_float_new_from_ut32_as_f32(0x44801003);
+	RzFloat *expect2 = rz_float_new_from_ut32_as_f32(0xC3F52F40);
 	RzFloat *rem2 = rz_float_rem(a2, b2, RZ_FLOAT_RMODE_RNE);
 	mu_assert_true(is_equal_float(rem2, expect2), "rem test 2");
 	rz_float_free(a2);
@@ -573,9 +573,9 @@ bool f32_ieee_rem_test(void) {
 	rz_float_free(expect2);
 	rz_float_free(rem2);
 
-	RzFloat *a3 = rz_float_new_from_hex_as_f32(0x3F7FFF3F);
-	RzFloat *b3 = rz_float_new_from_hex_as_f32(0x957CE0B6);
-	RzFloat *expect3 = rz_float_new_from_hex_as_f32(0x145F53B0);
+	RzFloat *a3 = rz_float_new_from_ut32_as_f32(0x3F7FFF3F);
+	RzFloat *b3 = rz_float_new_from_ut32_as_f32(0x957CE0B6);
+	RzFloat *expect3 = rz_float_new_from_ut32_as_f32(0x145F53B0);
 	RzFloat *rem3 = rz_float_rem(a3, b3, RZ_FLOAT_RMODE_RNE);
 	mu_assert_true(is_equal_float(rem3, expect3), "rem test 3");
 	rz_float_free(a3);

